### PR TITLE
[python,c++] Use C++ `DenseNDArray` bindings in Python API

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue and/or context:** 
https://github.com/single-cell-data/TileDB-SOMA/issues/2429

### CODE REVIEW

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 1]
```diff
@@ -1785,6 +1785,9 @@ def _write_matrix_to_denseNDArray(
             )
             return
 
+    if isinstance(matrix, sp.csc_matrix):
+        matrix = sp.csr_matrix(matrix)
```
Comment (nguyenv) [inline apis/python/src/tiledbsoma/io/ingest.py:1789]: This is a workaround for getting CSC matrices to write properly with the new `clib.DenseNDArray` bindings. I would appreciate it if someone could suggest a better solution.

**Review by johnkerl — CHANGES_REQUESTED** [review]
https://github.com/single-cell-data/TileDB-SOMA/pull/2494#discussion_r1584926419

File: apis/python/src/tiledbsoma/io/ingest.py (hunk 1 above)
Comment (nguyenv) [reply apis/python/src/tiledbsoma/io/ingest.py:1789]: Thank you so much @johnkerl! This is extremely helpful, and I really appreciate it. I couldn't figure it out after a weekend of debugging.

**Review by johnkerl — APPROVED** [review]
🚢 
Awesome @nguyenv !! :)

**Pull request merged**